### ISSUE TITLE
Remove height setting from `progress-bar`

### ISSUE THREAD

**Issue by DNin01** [ScratchAddons/ScratchAddons#5985]
### Why this would be helpful

Most users don’t care to change it.

Also, this addon was [recently updated](https://github.com/ScratchAddons/ScratchAddons/pull/5583) to change the design of the progress bar and the default height value, but I think we forgot to include migration code, so old users will still have the height set to 5 instead of the new default, 8. This change would effectively auto-update the height as there would no longer be a setting to change it, so it would become 8 pixels wide.

Reducing the number of customizations can also improve the user experience.

### My suggestion

Remove the progress bar height setting and always make it 8 pixels wide, the current default.

**Comment by Samq64** [comment]
I'd say just add the migration code now and leave the setting. If we want to start decluttering addon settings, something like Infinite scrolling or Scratch notifier would be a better place to start.

**Comment by BroJac5246** [comment]
I didn't include migration code because:
A. I didn't really think of it
B. I didn't want to change it for users who preferred 5px (realistically probably no one, but yeah)

I am personally for removing the setting altogether because who actually uses it? No one.

> If we wanted to start decluttering addon settings, addons like infinite scrolling, scratch notifier, or even themes like editor dark mode would be a much better place to start.

I would disagree. Sometimes it's good to start small, and if we declutter then it doesn't matter where we start, so long as it gets done. You do make a valid point about decluttering them in general, though.

**Comment by WorldLanguages** [comment]
> I think we should leave the setting add the migration code now. If we wanted to start decluttering addon settings, addons like infinite scrolling, scratch notifier, or even themes like editor dark mode would be a much better place to start.

+1 on trying to revamp the "infinite scrolling" addon! Can't believe "fix footer" is still enabled by default on it

**Comment by WorldLanguages** [comment]
This setting was added back in September 2020, together with the whole (new, at that point) addon: https://github.com/ScratchAddons/ScratchAddons/pull/153/files#diff-e9e6998dba9f7a7af6f1a3307bc3fe827b490e13f545d5c4489865ffea607166R30

Possibly it was motivated by @Hans5958's comment back then:
> Just a suggestion, how about making it thinner, like 2px?

**Comment by Hans5958** [comment]
Unrelated to the ping:

I personally disagree because of the fact that I want to give customization on some parts. Granted that I'm more of a top-of-the-page guy which prefers a slim progress bar. Also, it reduces the setting from 2 to 1, which is quite insigificant to change compared to other addons.

**Comment by Secret-chest** [comment]
Really, it's not hard to understand and it does no harm. Don't remove settings for the sake of it.

**Comment by Secret-chest** [comment]
> I think we should leave the setting add the migration code now. If we wanted to start decluttering addon settings, addons like infinite scrolling, scratch notifier, or even themes like editor dark mode would be a much better place to start.

Settings groups would be very useful in this regard. But we can't simplify themes (actually we need more settings for things like the cyan)

**Comment by WorldLanguages** [comment]
There's probably not enough consensus to remove the setting. So what migration are you suggesting?

**Comment by DNin01** [comment]
> There's probably not enough consensus to remove the setting. So what migration are you suggesting?

The migration I was proposing would've updated the progress bar's height to the new default, 8 pixels, after the aforementioned PR updated the design. Without it, new users would see a default height of 8 pixels as intended, but previous users would continue to see a height of 5 pixels.

Should that be its own issue?

**Comment by DNin01** [comment]
> Really, it's not hard to understand and it does no harm. Don't remove settings for the sake of it.

You're right, this setting doesn't really get in anyone's way at all, and the removal of this setting kind of contradicts our focus on customizability.

**Comment by WorldLanguages** [comment]
> The migration I was proposing would've updated the progress bar's height to the new default, 8 pixels, after the aforementioned PR updated the design. Without it, new users would see a default height of 8 pixels as intended, but previous users would continue to see a height of 5 pixels.

Yeah, but it's too late now to have a perfect migration. Best we can do now is to change everyone who has the addon disabled and set to 5px to 8px

**Comment by iqnite** [comment]
Why remove something that makes Scratch Addons more customizable? The mission of SA is to make Scratch customizable, so it should be, too. People who changed it earlier might be stuck in a size they don't like and won't be able to change it. And I'm sure that the day after this would get merged a new PR named "Re-add custom progress bar height" would pop up.
TL;DR: no.

**Comment by WorldLanguages** [comment]
> Why remove something that makes Scratch Addons more customizable?

Sometimes there's valid reasons. The more settings the harder to maintain. For example, in this case, we forgot to apply a setting migration and now we're having to spend time discussing this.

**Comment by iqnite** [comment]
Setting migration?

**Comment by DNin01** [comment]
> Setting migration?

Yes. The act of changing settings for users who are updating Scratch Addons (for example, updating a customizable color to a new default).
